src/zcash/JoinSplit.cpp

```diff
@@ -151,6 +144,7 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
 
     bool verify(
         const ZCProof& proof,
+        ProofVerifier& verifier,
```

Comment:
After reviewing all the diffs, I believe this parameter should use the ``const`` keyword because logically a verifier is immutable and concretely it's ``check`` method is itself ``const`` so the result should correctly compile.